gestalt/provider/vault.py

```diff
@@ -0,0 +1,67 @@
+from .provider import Provider
+import requests
+from typing import Optional, Tuple
+import hvac  # type: ignore
+import os
+
+
+class Vault(Provider):
+    def __init__(self,
+                 cert: Optional[Tuple[str, str]] = None,
+                 role: Optional[str] = None,
+                 jwt: Optional[str] = None,
+                 url: Optional[str] = os.environ.get("VAULT_ADDR"),
+                 token: Optional[str] = os.environ.get("VAULT_TOKEN"),
+                 verify: Optional[bool] = True) -> None:
+        """Initialized vault client and authenticates vault
+
+        Args:
+            client_config (HVAC_ClientConfig): initializes vault. URL can be set in VAULT_ADDR
+                environment variable, token can be set to VAULT_TOKEN environment variable.
+                These will be picked by default if not set to empty string
+            auth_config (HVAC_ClientAuthentication): authenticates the initialized vault client
+                with role and jwt string from kubernetes
+        """
+        self.vault_client = hvac.Client(url=url,
+                                        token=token,
+                                        cert=cert,
+                                        verify=verify)
+        try:
+            self.vault_client.is_authenticated()
+        except requests.exceptions.MissingSchema:
+            raise RuntimeError(
+                "Gestalt Error: Incorrect VAULT_ADDR or VAULT_TOKEN provided")
+        if role and jwt:
+            try:
+                self.vault_client.auth_kubernetes(role=role, jwt=jwt)
+            except hvac.exceptions.InvalidPath:
+                raise RuntimeError(
+                    "Gestalt Error: Kubernetes auth couldn't be performed")
+            except requests.exceptions.ConnectionError:
+                raise RuntimeError("Gestalt Error: Couldn't connect to Vault")
+
+    def get(self, key: str, path: str, filter: str) -> str:
```

Comment:
Will we always have a key when we run `.get()`

Reply:
Not sure if we want to switch that behaviour to return empty string back 🤔